tests/unit/test_csv_table_from_file.py

```diff
@@ -1,123 +1,89 @@
 import pytest
-from mock import Mock, patch, mock_open
-from behave.model import Feature, ScenarioOutline, Table, Row
 import os
-import logging
-
+import csv
+from mock import MagicMock, patch
+from os.path import dirname, abspath, join
 from behave.contrib.csv_table_from_file import read_examples_table_data_from_csv, \
-    preprocess_and_read_examples_table_data_from_csv
+    preprocess_and_read_examples_table_data_from_csv, logger
+from behave.model import Table, Feature, Examples
 
 
-class Example:
-    def __init__(self):
-        self.table = None
+@pytest.fixture
+def mock_example():
```

Comment:
Mmh, you rely too much on mocks.

* Therefore, you test something what you think is true, but the reality differs.
* It is much simpler to use the existing model classes that is used by the real code.
* See comment above `tag.startswith("@from_file=")`

DEFICIENCY:
* AFAIK by scanning over your tests: There is no test that really checks the functionality works as expected

Reply:
Resolved